File: types/collection.ts
export interface CollectionSchemaBase {
  collectionId: number;
  count: number;
  name: string;
  shortName: string;
  representation: string[]; // JSON parsed into an array of strings
  aspect1: string[];
  aspect2: string[];
  aspect3: string[];
}

export interface CollectionSchemaFull extends CollectionSchemaBase {
  createdAt: string;
  updatedAt: string;
}

File: schemas/collection.py
from datetime import datetime
from typing import List

from pydantic import BaseModel


class CollectionSchemaBase(BaseModel):
    collection_id: int
    topic_id: int  # Unique topic_id
    count: int
    name: str
    short_name: str
    representation: List[str]
    aspect1: List[str] # Same assumption for these JSON fields
    aspect2: List[str]
    aspect3: List[str]

    model_config = {
        "from_attributes": True
    }

class CollectionSchemaFull(CollectionSchemaBase):
    created_at: datetime
    updated_at: datetime
